packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -36,6 +37,8 @@ function HeaderToolbar() {
 	const inserterButton = useRef();
 	const { setIsInserterOpened, setIsListViewOpened } =
 		useDispatch( editPostStore );
+	const { get: getPreference } = useSelect( preferencesStore );
+	const hasFixedToolbar = getPreference( 'core/edit-post', 'fixedToolbar' );
```

Comment:
This isn't a good way to use selectors when value is used during the render. When the `fixedToolbar` preference value changes, this won't rerender the component.

Let's make it part of the `useSelect` below.

Reply:
I made a PR for this: https://github.com/WordPress/gutenberg/pull/52332

Reply:
There's another one in `packages/edit-site/src/components/header-edit-mode/index.js`, let fix both in one go :)

Reply:
done